enterprise/server/tasksize/tasksize.go

```diff
@@ -64,20 +86,140 @@ func Register(env environment.Env) error {
 	return nil
 }
 
-type taskSizer struct{}
+type taskSizer struct {
+	env environment.Env
+	rdb redis.UniversalClient
+}
 
 func NewSizer(env environment.Env) (*taskSizer, error) {
-	return &taskSizer{}, nil
+	var rdb redis.UniversalClient
+	if *useMeasuredSizes {
+		rdb = env.GetRemoteExecutionRedisClient()
+		if rdb == nil {
+			return nil, status.FailedPreconditionError("missing Redis client configuration")
+		}
+	}
+	return &taskSizer{
+		env: env,
+		rdb: rdb,
+	}, nil
```

Comment:
idk if this is better but you could skip the rdb var if you just assign directly into your taskSizer struct

	ts := &taskSizer{
		env: env,
	}
	if *useMeasuredSizes {
		if env.GetRemoteExecutionRedisClient() == nil {
			return nil, status.FailedPreconditionError("missing Redis client configuration")
		}
        ts.rdb = env.GetRemoteExecutionRedisClient()
	}
	return ts, nil